### DISCUSSION TITLE
Vercel Generative UI through function calling cannot set maxDuration

### DISCUSSION THREAD

**Discussion by allenchuang** [vercel/community#6644] [Help]
### Summary

I have cloned the most recent demo of the https://github.com/vercel/ai-chatbot

within the function call i'm trying to call using a fetch call that returns a response in 15+ seconds. However, i keep getting the message `Task timed out after 15.01 seconds`, which is extremely frustrating.

**UPDATE:** so it turns out that the timeout isn't from the api/ route as under the functions i see that maxDuration has been increased to 5 min, but rather from the `api.openai.com/v1/chat/completions` anyway we can get around that?

### Example
![image](https://github.com/vercel/community/assets/4248780/1fd5d8d9-c6c5-4bb6-a311-f1bdfc0e109c)
![image](https://github.com/vercel/community/assets/4248780/38bcc32d-7535-4600-8a8f-c65d310c82b6)

_No response_

### Steps to Reproduce

I have cloned the most recent demo of the https://github.com/vercel/ai-chatbot

within the function call i'm trying to call using a fetch call that returns a response in 15+ seconds. However, i keep getting the message `Task timed out after 15.01 seconds`, which is extremely frustrating.

I started out making a function that calls an external API directly and setting `maxDuration` there only to find out that in order to set maxDuration it has to be within `app/api/` folder. (see commented code)

So I made a route (which makes the call takes even longer) and even with `export const maxDuration=300` it is still failing after 15 sec

```
 getTopStores: {
        description:
          'Get the top stores.',
        parameters: z.object({
          type_of_food: z
            .string()
            .describe(
              'The type of restaurant or cusine, for example, pizza, sushi, burgers, fried chicken and more...'
            ),
          limit: z.number().describe('The number of restaurants to return')
        }),

        render: async function* ({ type_of_food, limit = 5 }) {
          yield (
            <BotCard>
              <EventsSkeleton />
            </BotCard>
          )
          // const restaurantList = await get_top_stores(
          //   type_of_food,
          //   limit
          // )

          const storeRes = await fetch(
            `${process.env.NEXT_API_URL}/stores/search/?storeUrl=${type_of_food}&limit=${limit}`
          )
          const storeList = await storeRes.json()

          aiState.done({
            ...aiState.get(),
            messages: [
              ...aiState.get().messages,
              {
                id: nanoid(),
                role: 'function',
                name: 'getTopStores',
                content: JSON.stringify(storeList)
              }
            ]
          })

          return (
            <BotCard>
              <RestaurantList restaurantList={restaurantList} />
            </BotCard>
          )
        }
      },
```

Can someone please help me?

Thanks in advance!

**Comment by aa-wong** [comment]
I found this response. @allenchuang have you tried this?

https://github.com/vercel/ai/issues/1255#issuecomment-2057872826
